### PR TITLE
:sparkles: display dropdown menu according to a condition

### PR DESCRIPTION
**Pull request by azertyValentin**
To use the new dropdown condition, you can add the following line to the projects settings which uses the 2FA:

ADMIN_DISPLAY_DROP_DOWN_MENU_CONDITION_FUNCTION = lambda request: request.user.is_verified()

### CODE REVIEW

**Review by annashamray — CHANGES_REQUESTED** [review]
I don't think this is a right place for this change

If dropdown should not be displayed in the login page, the login page template in the 2FA lib should be modified. 

`admin_index` should not rely on 2FA environment variables

**Comment by azertyValentin** [comment]
> I don't think this is a right place for this change
> 
> If dropdown should not be displayed in the login page, the login page template in the 2FA lib (or in your specific project) should be modified.
> 
> `admin_index` should not rely on 2FA environment variables

I agree, my last commit fix this issue

File: django_admin_index/templatetags/django_admin_index.py [hunk 1]
```diff
@@ -32,3 +32,8 @@ def dashboard_app_list(context):
 @register.simple_tag()
 def admin_index_settings():
     return settings.as_dict()
+
+
+@register.simple_tag
+def display_dropdown_menu(request):
+    return settings.DISPLAY_DROP_DOWN_MENU_CONDITION_FUNCTION(request)
```
Comment (sergei-maertens) [inline django_admin_index/templatetags/django_admin_index.py:39]: The setting should be a string value, something like `"django_admin_index.utils.should_display_dropdown_menu"`.

You can then import this function using `django.utils.module_loading.import_string` and invoke it:

```suggestion
def display_dropdown_menu(request):
    func = import_string(settings.DISPLAY_DROP_DOWN_MENU_CONDITION_FUNCTION)
    return func(request)
```

This makes it possible to pass the function to use as an environment variable.

File: django_admin_index/conf.py [hunk 2]
```diff
@@ -3,6 +3,9 @@
 from django.conf import settings as django_settings
 
 
+def default_dropdown_condition_function(parameter) -> bool:
+    return True
+
```
Comment (sergei-maertens) [inline django_admin_index/conf.py:8]: Please move this to `django_admin_index.utils.should_display_dropdown_menu`, and change the implementation to contain the code that is now in the template:

```py
def should_display_dropdown_menu(request):
    return admin_index_settings.SHOW_MENU and request.user.is_authenticated and request.user.is_staff
```

File: django_admin_index/templates/admin/base.html [hunk 3]
```diff
@@ -64,26 +64,29 @@
...
-                <span>{{ app.name }}</span>
-                <span class="dropdown-menu__drop">
-                    {% for model in app.models %}
-                        {% if model.admin_url %}
-                            <div class="dropdown-menu__drop-item">
-                                <a href="{{ model.admin_url }}" class="dropdown-menu__link {% if model.active %}dropdown-menu__link--active{% endif %}">{{ model.name }}</a>
-                            </div>
-                        {% endif %}
+        {% display_dropdown_menu request as display_dropdown %}
+        {% if display_dropdown == True %}
```
Comment (sergei-maertens) [inline django_admin_index/templates/admin/base.html:68]: Using the above changes in the functions, please simplify the template:

```django
{% display_dropdown_menu request as should_display_dropdown %}
{% if should_display_dropdown and dashboard_app_list %}
...
```

**Review by sergei-maertens — CHANGES_REQUESTED** [review]

**Review by sergei-maertens — APPROVED** [review]

**Pull request merged**